04-store-api/starter/app.js

```diff
@@ -1 +1,37 @@
-console.log('04 Store API')
+require("dotenv").config()
+require("express-async-errors")
+const connectDB = require("./db/connect")
+const productsRouter = require("./routes/products")
+const PORT = process.env.PORT || 3000
+
+const express = require("express")
+const app = express()
+
+const notFoundMiddleware = require("./middleware/not-found")
+const errorMiddleware = require("./middleware/error-handler")
+
+// middleware
+app.use(express.json())
+
+// routes
+app.get("/", (req, res) => {
```

Comment:
extra challenge: what if we returned a machine-readable JSON object at the root route instead of a html page? How would we do that?